packages/blocks/src/api/utils.js

```diff
@@ -371,7 +372,13 @@ export const __experimentalGetBlockAttributesNamesByRole = ( ...args ) => {
 };
 
 export function isContentBlock( name ) {
-	const attributes = getBlockType( name )?.attributes;
+	const blockType = getBlockType( name );
+
+	if ( blockType?.[ blockMetadataDisplayKey ]?.role === 'content' ) {
```

Comment:
This is the special sauce. 

There are probably other scenarios where this key needs to be checked as well.